lib/note-editor.jsx

```diff
@@ -168,9 +163,16 @@ export const NoteEditor = React.createClass( {
 	}
 } );
 
-const mapStateToProps = ( { settings } ) => ( {
+const mapStateToProps = ( { revisions, settings } ) => ( {
 	fontSize: settings.fontSize,
-	markdownEnabled: settings.markdownEnabled
+	markdownEnabled: settings.markdownEnabled,
+	revisionState: revisions.revisionState,
+} );
+
+const mapDispatchToProps = dispatch => ( {
+	onRevisionChange: revisionState => {
+		dispatch( setRevisionState( revisionState ) );
+	},
 } );
```

Comment:
what if we got rid of the more generic names moving forward in our state system and used names that more accurately describe what we are doing.

in this case, the component isn't actually changing anything - it's simply indicating the type of change it wants to see in the world. in fact, that action is inherently coupled to the meaning of the component, in which case there's little to no need to abstract it away (and runaway prop-passing abstraction is one of the big weaknesses for this app in my opinion)

instead, let's name it for what it is

```js
const mapDispatchToProps = dispatch => ( {
	setRevision: revision => dispatch( setRevisionState( revision ) ),
	unsetRevision: () => dispatch( setRevisionState( null ) ),
} );
```

and now the component doesn't need to know anything about the mechanics of how this works or how to clear. it API is simplified to two functions: `set( revision )` and `unset()`

if we want to think differently, we can compose…

```js
const mapDispatchToProps = dispatch => ( {
	setRevision: compose( dispatch, setRevisionState ),
	unsetRevision: partial( dispatch, partial( setRevisionState, null ) ),
} );
```